src/adapters/surrealDB/types/base.ts

```diff
@@ -0,0 +1,86 @@
+import type { BaseResponse } from '../../../types';
+
+export type SurrealDbResponse = {} & BaseResponse;
+
+export type EnrichedBqlQueryRelation = {
+	$thingType: 'relation';
+	$plays: string;
+	$playedBy: {
+		path: string;
+		relation: string;
+		plays: string;
+		target: string;
+		thing: string;
+		thingType: string;
+	};
+	$path: string;
+	$dbPath: undefined;
+	$as: string;
+	$var: string;
+	$thing: string;
+	$fields: Array<any>;
+	$excludedFields: undefined;
+	$fieldType: string;
+	$target: string;
+	$intermediary: string;
+	$justId: boolean;
+	$id: undefined;
+	$filter: {};
+	$idNotIncluded: boolean;
+	$filterByUnique: boolean;
+	$filterProcessed: boolean;
+};
+
+export type EnrichedBqlQueryEntity = {
```

Comment:
We could bstract common props  as `EnrichedBQLQueryThingProps` to avoid repeating them